examples/sycl/pvc/pvc_gemm.cpp

```diff
@@ -170,95 +170,115 @@ struct ExampleRunner {
 
   cutlass::DeviceAllocation<ElementA> block_A;
   cutlass::DeviceAllocation<ElementB> block_B;
-  cutlass::DeviceAllocation<ElementB> block_B_vnni;
-  cutlass::DeviceAllocation<ElementC> block_C;
+  // cutlass::DeviceAllocation<ElementC> block_C;
   cutlass::DeviceAllocation<ElementOutput> block_D;
   cutlass::DeviceAllocation<ElementOutput> block_ref_D;
 
+  static auto constexpr l3_cache_size = 256 * 1024 * 1024;
+
+  size_t PINGPONG_ITER = 1;
+  size_t pingpong_size_a;
+  size_t pingpong_size_b;
+  size_t pingpong_size_d;
+
+  std::vector<ElementA> a;
+  std::vector<ElementB> b;
+  std::vector<ElementC> d;
   //
   // Methods
   //
 
-  bool verify(const ProblemShapeType& problem_size, ElementCompute alpha, ElementCompute beta) {
+  bool verify(ProblemShapeType const& problem_size, ElementCompute alpha, ElementCompute beta) {
     auto [M, N, K, L] = problem_size;
 
     cutlass::TensorRef ref_A(block_A.get(), LayoutA::packed({M, K}));
     cutlass::TensorRef ref_B(block_B.get(), LayoutB::packed({K, N}));
-    cutlass::TensorRef ref_C(block_C.get(), LayoutC::packed({M, N}));
+    cutlass::TensorRef ref_C((ElementC*)nullptr /*block_C.get()*/, LayoutC::packed({M, N}));
     cutlass::TensorRef ref_D(block_ref_D.get(), LayoutD::packed({M, N}));
 
     cutlass::reference::device::GemmComplex(
-          {M, N, K},
-          alpha,
-          ref_A,
-          cutlass::ComplexTransform::kNone,
-          ref_B,
-          cutlass::ComplexTransform::kNone,
-          beta,
-          ref_C,
-          ref_D,
-          ElementAccumulator(0),
-          L,     // batch_count
-          M * K, // batch_stride_A
-          K * N, // batch_stride_B
-          M * N, // batch_stride_C
-          M * N  // batch_stride_D
-        );
+        {M, N, K}, alpha, ref_A, cutlass::ComplexTransform::kNone, ref_B,
+        cutlass::ComplexTransform::kNone, beta, ref_C, ref_D, ElementAccumulator(0),
+        L,     // batch_count
+        M * K, // batch_stride_A
+        K * N, // batch_stride_B
+        M * N, // batch_stride_C
+        M * N  // batch_stride_D
+    );
 
     syclcompat::wait();
 
-    // Check if output from CUTLASS kernel and reference kernel are relatively equal or not
-    // need to set a larger error margin for comparison to succeed
-    auto epsilon = static_cast<ElementOutput>(0.1f);
-    auto nonzero_floor = static_cast<ElementOutput>(0.1f);
-
+    // Check if output from CUTLASS kernel and reference kernel are relatively
+    // equal or not need to set a larger error margin for comparison to succeed
     bool passed = cutlass::reference::device::BlockCompareRelativelyEqual(
-            block_ref_D.get(), block_D.get(), block_D.size(),
-            epsilon, nonzero_floor);
+        block_ref_D.get(), block_D.get(), M * N * L, 0.5f, 0.5f);
```

Comment:
Hi, We found while _epsilon=0.1f, nonzero_floor=0.1f_ if require that if diff < 0.1f then need diff < 0.01f. That means
val = 2000.1 and ref_val = 2000 cannot pass the verification.  Do you have any suggestion here?
[https://github.com/codeplaysoftware/cutlass-fork/blob/fae9b2e19f799ba7ffbadec70d3ce9bd1e2701ed/tools/util/include/cutlass/util/reference/device/tensor_compare.h#L103](BlockCompareRelativelyEqual) [https://github.com/codeplaysoftware/cutlass-fork/blob/fae9b2e19f799ba7ffbadec70d3ce9bd1e2701ed/include/cutlass/relatively_equal.h#L73](relatively_equal)

Reply:
Was the bump in tolerance value required for GEMM + Epilogue or even just GEMM as well ?

Reply:
Hi @Jiaxingla,  I think the issue is in how the reference value is calculated. The current implementation is doing something equivalent to: 

```c++
float accum[kMblock][kNblock];

..... 

bfloat16_t a = tensor_a.at(MatrixCoord(row, k_block));
bfloat16_t b = tensor_b.at(MatrixCoord(k_block, col));

float a_ik = float(a);
float b_kj = float(b);

accum[i][j] = float(a_ik) * float(b_kj) + accum[i][j];
```

This matches what Nvidia is doing in the MMA operation. However for Intel, I think the built-in function is doing something more like:

```c++
float accum[kMblock][kNblock];

..... 

bfloat16_t a = tensor_a.at(MatrixCoord(row, k_block));
bfloat16_t b = tensor_b.at(MatrixCoord(k_block, col));

accum[i][j] = bfloat16_t(a) * bfloat16_t(b) + accum[i][j];
```

I can't confirm this as I'm struggling to find documentation about the built-in functions, but it would explain the accumulation error that we are seeing.

The gemm reference implementation is in [here](https://github.com/codeplaysoftware/cutlass-fork/blob/fae9b2e19f799ba7ffbadec70d3ce9bd1e2701ed/tools/util/include/cutlass/util/reference/device/gemm_complex.h#L155 ) and the `inner_product_op` is [this one](https://github.com/codeplaysoftware/cutlass-fork/blob/fae9b2e19f799ba7ffbadec70d3ce9bd1e2701ed/include/cutlass/functional.h#L439)